src/LuaAST.ts

```diff
@@ -438,6 +441,22 @@ export function createLabelStatement(name: string, tsOriginal?: ts.Node, parent?
     return statement;
 }
 
+
+export interface AwaitExpression extends Expression {
+    kind: SyntaxKind.AwaitExpression;
+    expression: UnaryExpression;
+}
+
+export function isAwaitExpression(node: Node): node is AwaitExpression {
+  return node.kind === SyntaxKind.AwaitExpression;
+}
+
+export function createAwaitExpression(expression: Expression): AwaitExpression {
+  const expr = createNode(SyntaxKind.AwaitExpression ) as AwaitExpression;
+  setParent(expression, expr);
+  return expr;
+}
+
```

Comment:
Please dont modify the AST.
This is a Lua AST implementation, the idea is to have no TS related stuff in here.
As fas as I can tell you dont even use the added node.

Reply:
You are absolutely right of cause. I removed all modifications to the lua ast